app/gui/suggestion-database/src/documentation_ir.rs

```diff
@@ -152,24 +164,26 @@ pub enum Placeholder {
 pub enum Documentation {
     Module(Rc<ModuleDocumentation>),
     Type(Rc<TypeDocumentation>),
-    Constructor {
-        name:      Rc<QualifiedName>,
-        type_docs: Rc<TypeDocumentation>,
-    },
-    Method {
-        name:      Rc<QualifiedName>,
-        type_docs: Rc<TypeDocumentation>,
-    },
-    ModuleMethod {
-        name:        Rc<QualifiedName>,
-        module_docs: Rc<ModuleDocumentation>,
-    },
-    /// We temporarily do not support documentation of the `Local` entries because the language
-    /// server does not provide us with any. See https://www.pivotaltracker.com/story/show/183970215.
-    Function {},
-    /// We temporarily do not support documentation of the `Local` entries because the language
-    /// server does not provide us with any. See https://www.pivotaltracker.com/story/show/183970215.
-    Local {},
+    Constructor { name: Rc<QualifiedName>, type_docs: Rc<TypeDocumentation> },
+    Method { name: Rc<QualifiedName>, type_docs: Rc<TypeDocumentation> },
+    ModuleMethod { name: Rc<QualifiedName>, module_docs: Rc<ModuleDocumentation> },
+    Function(Rc<FunctionDocumentation>),
+    Local(Rc<LocalDocumentation>),
+}
+
+impl Documentation {
+    /// Qualified name of the documented function. Functions are part of the documentation for
+    /// the larger entity, e.g., constructor documentation is embedded into the type
+    /// documentation. The returned qualified name is used to scroll to the corresponding section in
+    /// a larger documentation page.
+    pub fn function_name(&self) -> Option<&QualifiedName> {
+        match self {
+            Documentation::Constructor { name, .. } => Some(name),
+            Documentation::Method { name, .. } => Some(name),
+            Documentation::ModuleMethod { name, .. } => Some(name),
```

Comment:
use `self.name` - less refactoring needd if we ever change the var "name".

Reply:
We can't really do that, if I understand you correctly. You can't access field of the enum without pattern matching.